aws/resource_aws_cloudwatch_event_bus_policy.go

```diff
@@ -0,0 +1,179 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	tfevents "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/cloudwatchevents"
+	iamwaiter "github.com/terraform-providers/terraform-provider-aws/aws/internal/service/iam/waiter"
+)
+
+func resourceAwsCloudWatchEventBusPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudWatchEventBusPolicyCreate,
+		Read:   resourceAwsCloudWatchEventBusPolicyRead,
+		Update: resourceAwsCloudWatchEventBusPolicyUpdate,
+		Delete: resourceAwsCloudWatchEventBusPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("event_bus_name", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"event_bus_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCloudWatchEventBusName,
```

Comment:
This was updated in latest releases to also take into account Partner buses that slightly differ in ARN.

```suggestion
				ValidateFunc: validateCloudWatchEventBusNameOrARN,
```